Kitodo/src/main/java/org/kitodo/production/services/command/MetadataScript.java

```diff
@@ -12,60 +12,62 @@
 package org.kitodo.production.services.command;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
+import org.apache.commons.lang.text.StrTokenizer;
+
 public class MetadataScript {
 
-    private String goal;
-    private String root;
+    private Map<String, String> parameters;
+    private String metadataKey;
+    private String valueSource;
     private String typeTarget;
+    private String variable;
     private List<String> values = new ArrayList<>();
 
     /**
      * Creates a MetadataScript with given command.
      * @param command the given command.
      */
     public MetadataScript(String command) {
-        String metadataCommand = command;
-        String typeCommand = "";
-        if (command.contains("+")) {
-            String[] commandParts = command.split("\\+");
-            metadataCommand = commandParts[0];
-            typeCommand = commandParts[1];
-        }
-        if (metadataCommand.contains("=")) {
-            String[] metadataParts = metadataCommand.split("=");
-            goal = metadataParts[0];
-            String rootOrValue = metadataParts[1];
-            if (rootOrValue.startsWith("@") || rootOrValue.startsWith("$")) {
-                root = rootOrValue;
-            } else {
-                values.add(rootOrValue);
+        this.parameters = new HashMap<>();
+        StrTokenizer tokenizer = new StrTokenizer(command, ' ', '\"');
+        while (tokenizer.hasNext()) {
```

Comment:
The script is already tokenized in the `KitodoScriptService` class. The result is available there in the class field `parameters` of the class `KitodoScriptService`. Please do not tokenize the script here again, instead, pass the `parameters` field through here.

Reply:
It's not parsed correctly in the KitodoScriptService. I cannot use it, when multiple scripts are entered with ';'